File: javascript/perfect-numbers/perfect-numbers.js
export const classify = (number) => {
	if (number <= 0) { throw new Error('Classification is only possible for natural numbers.') }

	if (number == 1) { return "deficient"; }

	let numberFactors = [1];

	for (let i = 2; i < number; i++) {
		if (number % i == 0 && !numberFactors.includes(i)) {
			numberFactors.push(i)
		}
	}

	let sumOfFactors = numberFactors.reduce((accumulator, currentValue) => accumulator + currentValue, 0);

	if (sumOfFactors === number) {
		return "perfect";
	}
	else if (sumOfFactors > number) {
		return "abundant";
	}
	else {
		return "deficient";
	}
};

File: csharp/perfect-numbers/PerfectNumbers.cs
using System;
using System.Collections.Generic;
using System.Linq;

public enum Classification
{
    Perfect,
    Abundant,
    Deficient=-1
}

public static class PerfectNumbers
{
    public static Classification Classify(int number)
    {
        if (number < 1) { throw new ArgumentOutOfRangeException(); }

        return (Classification)(Enumerable.Range(1, number / 2)
                                       .Where(x => number % x == 0)
                                       .Sum()).CompareTo(number);
    }
}
